authentication/tests/test_user_login.py

```diff
@@ -0,0 +1,62 @@
+from rest_framework import status
+from rest_framework.test import APITestCase
+
+from authentication.models import CustomUser
+
+
+class UserLoginAPITests(APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.test_user = CustomUser.objects.create_user(
+            person_email="test@test.com",
+            password="Test1234",
+            person_name="Test",
+            person_surname="Test",
+        )
+        cls.test_user.is_active = True
+        cls.test_user.save()
+
+    def test_login_successful(self):
+        url = "/api/auth/token/login/"
+        data = {
+            "person_email": "test@test.com",
+            "password": "Test1234",
+        }
+        response = self.client.post(url, data)
+        self.assertEqual(response.status_code,
+                         status.HTTP_200_OK)
+        self.assertContains(response, "auth_token")
+
+    def test_login_email_incorrect(self):
+        url = "/api/auth/token/login/"
+        data = {
+            "person_email": "tost@test.com",
+            "password": "Test1234",
+        }
+        response = self.client.post(url, data)
```

Comment:
The `url` and `data` variables are redundant.

I recommend to put explicit values directly in calls. It's a bit easier to read what's going on, when the content is in-place instead of being somewhere above in variables.

Also (please share it with ther rest of the team) — from my experience, unittests are the place where it's better to write as explicit code, as possible. Ignoring things as code duplication etc.

I found it very useful to keep unittests in the shape which is more or less readable by people who has no experience in BE, Python or even programming at all (and it _is_ possible to keep it this way).

Reply:
Redundant variables were removed from all the tests. 
Thank you for your recommendations a lot.